crates/ruff_python_formatter/src/builders.rs

```diff
@@ -36,7 +36,7 @@ pub struct LeadingComments<'a> {
 impl Format<ASTFormatContext<'_>> for LeadingComments<'_> {
     fn fmt(&self, f: &mut Formatter<ASTFormatContext<'_>>) -> FormatResult<()> {
         for comment in self.comments {
-            if matches!(comment.relationship, Relationship::Leading) {
+            if comment.relationship.is_leading() {
                 if let TriviaKind::OwnLineComment(range) = comment.kind {
```

Comment:
This is up to preference, but if you want, these changes are also possible.

```suggestion
                if let Some(range) = comment.kind.own_line_comment() {
```

Reply:
I prefer what we have now.